PWGJE/DataModel/EMCALClusters.h

```diff
@@ -17,34 +17,77 @@
+const EMCALClusterDefinition kV3Variation1(ClusterAlgorithm_t::kV3, 11, 1, "kV3Variation1", 0.1, 0.3, -10000, 10000, 0.03);
+const EMCALClusterDefinition kV3Variation2(ClusterAlgorithm_t::kV3, 12, 1, "kV3Variation2", 0.1, 0.2, -10000, 10000, 0.03);
+
+/// \brief function returns EMCALClusterDefinition for the given name
+/// \param name name of the cluster definition
+/// \return EMCALClusterDefinition for the given name
+const EMCALClusterDefinition getClusterDefinitionFromString(const std::string& clusterDefinitionName)
+{
+  if (clusterDefinitionName == "kV1Default") {
+    return kV1Default;
+  } else if (clusterDefinitionName == "kV1Variation1") {
+    return kV1Variation1;
+  } else if (clusterDefinitionName == "kV1Variation2") {
+    return kV1Variation2;
+  } else if (clusterDefinitionName == "kV3Default") {
+    return kV3Default;
+  } else if (clusterDefinitionName == "kV3Variation1") {
+    return kV3Variation1;
+  } else if (clusterDefinitionName == "kV3Variation2") {
+    return kV3Variation2;
+  } else {
+    throw std::invalid_argument("Cluster definition name not recognized");
+  }
+};
+
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);                        //! collisionID used as index for matched clusters
+DECLARE_SOA_INDEX_COLUMN(BC, bc);                                      //! bunch crossing ID used as index for ambiguous clusters
+DECLARE_SOA_COLUMN(ID, id, int);                                       //! cluster ID identifying cluster in event
+DECLARE_SOA_COLUMN(Energy, energy, float);                             //! cluster energy (GeV)
+DECLARE_SOA_COLUMN(CoreEnergy, coreEnergy, float);                     //! cluster core energy (GeV)
+DECLARE_SOA_COLUMN(Eta, eta, float);                                   //! cluster pseudorapidity (calculated using vertex)
+DECLARE_SOA_COLUMN(Phi, phi, float);                                   //! cluster azimuthal angle (calculated using vertex)
+DECLARE_SOA_COLUMN(M02, m02, float);                                   //! shower shape long axis
+DECLARE_SOA_COLUMN(M20, m20, float);                                   //! shower shape short axis
+DECLARE_SOA_COLUMN(NCells, nCells, int);                               //! number of cells in cluster
+DECLARE_SOA_COLUMN(Time, time, float);                                 //! cluster time (ns)
+DECLARE_SOA_COLUMN(IsExotic, isExotic, bool);                          //! flag to mark cluster as exotic
+DECLARE_SOA_COLUMN(DistanceToBadChannel, distanceToBadChannel, float); //! distance to bad channel
+DECLARE_SOA_COLUMN(NLM, nlm, int);                                     //! number of local maxima
+DECLARE_SOA_COLUMN(Definition, definition, int);                       //! cluster definition, see EMCALClusterDefinition.h
 
+} // namespace emcalcluster
+// table of clusters that could be matched to a collision
 DECLARE_SOA_TABLE(EMCALClusters, "AOD", "EMCALCLUSTERS", //!
```

Comment:
I only noticed now that this should not be plural in the in-file table name. Can you adjust to "EMCALCLUSTER", please.

Reply:
@fjonasALICE